src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs

```diff
@@ -0,0 +1,268 @@
+using System.Windows.Automation.Peers;
+using System.Media;
+using System.IO;
+using Microsoft.Win32;
+using System.Reflection;
+using System.Timers;
+using System;
+using AccessibilityInsights.CommonUxComponents.Controls;
+using AccessibilityInsights.Win32;
+
+namespace AccessibilityInsights.CommonUxComponents.Controls
+{
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+    /// <summary>
+    /// Loading indicator modelled on UWP ProgressRing
+    /// </summary>
+    public partial class ProgressRingControl : UserControl
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+    {
+        /// <summary>
+        /// DependencyProperty IsActive
+        /// </summary>
+        public static readonly DependencyProperty IsActiveProperty =
+            DependencyProperty.Register(nameof(IsActive), typeof(bool), typeof(ProgressRingControl), 
+                new PropertyMetadata(new PropertyChangedCallback(IsActivePropertyChanged)));
+
+        /// <summary>
+        /// DependencyProperty WithSound
+        /// </summary>
+        public static readonly DependencyProperty WithSoundProperty =
+            DependencyProperty.Register(nameof(WithSound), typeof(bool), typeof(ProgressRingControl),
+                new PropertyMetadata(new PropertyChangedCallback(WithSoundPropertyChanged)));
+
+        /// <summary>
+        /// IsActive Property Change event handler
+        /// </summary>
+        /// <param name="d"></param>
+        /// <param name="e"></param>
+        private static void IsActivePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var control = d as ProgressRingControl;
+            if (control != null)
+            {
+                control.IsActive = (bool) e.NewValue;
+            }
+        }
+
+        /// <summary>
+        /// WithSound Property Change event handler
+        /// </summary>
+        /// <param name="d"></param>
+        /// <param name="e"></param>
+        private static void WithSoundPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var control = d as ProgressRingControl;
+            if (control != null)
+            {
+                control.WithSound = (bool)e.NewValue;
+            }
+        }
+
+        /// <summary>
+        /// Private variables used
+        /// </summary>
+        private SoundPlayer player;
+        private Timer timer;
+        private bool isPlayingSound;
+        private bool withSound;
+
+        /// <summary>
+        /// Check if Narrator is running
+        /// </summary>
+        public static bool IsInternalScreenReaderActive()
+        {
+            try
+            {
+                using (RegistryKey key = Registry.CurrentUser.OpenSubKey(Properties.Resources.ProgressRingControl_IsInternalScreenReaderActive_Software_Microsoft_Windows_NT_CurrentVersion_AccessibilityTemp))
```

Comment:
>ProgressRingControl_IsInternalScreenReaderActive_Software_Microsoft_Windows_NT_CurrentVersion_AccessibilityTemp [](start = 94, length = 111)

It was probably unintended to put a constant registry path into a resource string, but that's probably out of scope to fix here

Reply:
Yes and yes